### PR TITLE
Add docstrings

### PR DESCRIPTION
**Pull request by will-jl944**

### CODE REVIEW

File: paddlex/cv/models/base.py [hunk 1]
```diff
@@ -354,6 +354,15 @@ def analyze_sensitivity(self,
                             batch_size=8,
                             criterion='l1_norm',
                             save_dir='output'):
+        """
+
+        Args:
+            dataset: Dataset used for evaluation during sensitivity analysis.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/base.py]: (paddlex.dataset)

File: paddlex/cv/models/base.py [hunk 2]
```diff
@@ -354,6 +354,15 @@ def analyze_sensitivity(self,
...
+            batch_size(int): Batch size used in evaluation.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/base.py]: Default: 8.

File: paddlex/cv/models/classifier.py [hunk 3]
```diff
@@ -214,6 +235,17 @@ def train(self,
...
     def evaluate(self, eval_dataset, batch_size=1, return_details=False):
+        """
+        Evaluate the model.
+        Args:
+            eval_dataset(paddlex.dataset): Evaluation dataset.
+            batch_size(int, optional): Total batch size among all workers used for evaluation. Defaults to 1.
+            return_details(bool, optional): Whether to return evaluation details. Defaults to False.
+
+        Returns:
+            dict: Evaluation metrics.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/classifier.py]: 说明都有哪些key以及key代表的含义

File: paddlex/cv/models/classifier.py [hunk 4]
```diff
@@ -248,6 +280,17 @@ def evaluate(self, eval_dataset, batch_size=1, return_details=False):
...
     def predict(self, img_file, transforms=None, topk=1):
+        """
+        Do inference.
+        Args:
+            img_file(list or str): Input image paths or arrays in BGR format.
+            transforms(paddlex.transforms.Compose or None, optional): Transforms for inputs. If None, the transforms for evaluation process will be used. Defaults to None.
+            topk(int, optional): Keep topk results in prediction. Defaults to 1.
+
+        Returns:
+            dict or list of dict: The prediction results.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/classifier.py]: If `img_file` is a string or np.array, the result is a dict with key-value pairs: {"category_id": category_id, "category": category, "score":  score}. If `img_file` is a list, the result is a list composed of dicts.

With the corresponding fields:
category_id(int): the predicted category ID
category(str): category name
score(str): confidence

File: paddlex/cv/models/classifier.py [hunk 5]
```diff
@@ -248,6 +280,17 @@ def evaluate(self, eval_dataset, batch_size=1, return_details=False):
             return eval_metrics.get()
 
     def predict(self, img_file, transforms=None, topk=1):
+        """
+        Do inference.
+        Args:
+            img_file(list or str): Input image paths or arrays in BGR format.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/classifier.py]: img_file(list or str or np.array)：image path or decoded image data  in a BGR format, which also could constitute a list, means all images to be predicted at the same time.

File: paddlex/cv/models/classifier.py [hunk 6]
```diff
@@ -169,6 +168,28 @@ def train(self,
               early_stop=False,
               early_stop_patience=5,
               use_vdl=True):
+        """
+        Train the model.
+        Args:
+            num_epochs(int): The number of epochs.
+            train_dataset(paddlex.dataset): Training dataset.
+            train_batch_size(int, optional): Total batch size among all workers used in training. Defaults to 64.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/classifier.py]: workers -> cards

File: paddlex/cv/models/detector.py [hunk 7]
```diff
@@ -171,6 +171,29 @@ def train(self,
               early_stop=False,
               early_stop_patience=5,
               use_vdl=True):
+        """
+        Train the model.
+        Args:
+            num_epochs(int): The number of epochs.
+            train_dataset(paddlex.dataset): Training dataset.
+            train_batch_size(int, optional): Total batch size among all workers used in training. Defaults to 64.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/detector.py]: workers -> cards

File: paddlex/cv/models/classifier.py [hunk 8]
```diff
@@ -169,6 +168,28 @@ def train(self,
...
+            eval_dataset(paddlex.dataset, optional): Evaluation dataset. If None, the model will not be evaluated during training process. Defaults to None.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/classifier.py]: 太长了，换行写，行首顶4个空格

File: paddlex/cv/models/detector.py [hunk 9]
```diff
@@ -248,9 +271,21 @@ def train(self,
...
+        Evaluate the model.
+        Args:
+            eval_dataset(paddlex.dataset): Evaluation dataset.
+            batch_size(int, optional): Total batch size among all workers used for evaluation. Defaults to 1.
+            metric({'VOC', 'COCO', None}, optional): Evaluation metric. If None, determine the metric according to the dataset format. Defaults to None.
+            return_details(bool, optional): Whether to return evaluation details. Defaults to False.
+
+        Returns:
+            dict: Evaluation metrics.
+
```
Comment (FlyingQianMM) [inline paddlex/cv/models/detector.py:291]: 说明都有哪些key以及key代表的含义

File: paddlex/cv/models/detector.py [hunk 10]
```diff
@@ -347,6 +382,16 @@ def evaluate(self,
...
+        """
+        Do inference.
+        Args:
+            img_file(list or str): Input image paths or arrays in BGR format.
+            transforms(paddlex.transforms.Compose or None, optional): Transforms for inputs. If None, the transforms for evaluation process will be used. Defaults to None.
+
+        Returns:
+            dict or list of dict: The prediction results.
+
+        """
```
Comment (FlyingQianMM) [inline paddlex/cv/models/detector.py:407]: 说明都有哪些key以及key代表的含义

File: paddlex/cv/models/classifier.py [hunk 11]
```diff
@@ -169,6 +168,28 @@ def train(self,
...
+            optimizer(paddle.optimizer.Optimizer or None, optional): Optimizer used for training. If None, a default optimizer is used. Defaults to None.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/classifier.py]: 太长了，换行写，行首顶4个空格

File: paddlex/cv/models/classifier.py [hunk 12]
```diff
@@ -169,6 +168,28 @@ def train(self,
...
+            save_interval_epochs(int, optional): Epoch interval for saving the model. Defaults to 1.
+            log_interval_steps(int, optional): Step interval for printing training information. Defaults to 10.
+            save_dir(str, optional): Directory to save the model. Defaults to 'output'.
+            pretrain_weights(str or None, optional): None or name/path of pretrained weights. If None, no pretrained weights will be loaded. Defaults to 'IMAGENET'.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/classifier.py]: 太长了，换行写，行首顶4个空格

File: paddlex/cv/models/classifier.py [hunk 13]
```diff
@@ -248,6 +280,17 @@ def evaluate(self, eval_dataset, batch_size=1, return_details=False):
             return eval_metrics.get()
 
     def predict(self, img_file, transforms=None, topk=1):
+        """
+        Do inference.
+        Args:
+            img_file(list or str): Input image paths or arrays in BGR format.
+            transforms(paddlex.transforms.Compose or None, optional): Transforms for inputs. If None, the transforms for evaluation process will be used. Defaults to None.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/classifier.py]: 太长了，换行写，行首顶4个空格

File: paddlex/cv/models/detector.py [hunk 14]
```diff
@@ -171,6 +171,29 @@ def train(self,
...
+            save_interval_epochs(int, optional): Epoch interval for saving the model. Defaults to 1.
+            log_interval_steps(int, optional): Step interval for printing training information. Defaults to 10.
+            save_dir(str, optional): Directory to save the model. Defaults to 'output'.
+            pretrain_weights(str or None, optional): None or name/path of pretrained weights. If None, no pretrained weights will be loaded. Defaults to 'IMAGENET'.
+            learning_rate(float, optional): Learning rate for training. Defaults to .001.
+            warmup_steps(int, optional): The number of steps of warm-up training. Defaults to 0.
+            warmup_start_lr(float, optional): Start learning rate of warm-up training. Defaults to 0..
+            lr_decay_epochs(list or tuple, optional): Epoch milestones for learning rate decay. Defaults to (216, 243).
+            lr_decay_gamma(float, optional): Gamma coefficient of learning rate decay. Defaults to .1.
+            metric({'VOC', 'COCO', None}, optional): Evaluation metric. If None, determine the metric according to the dataset format. Defaults to None.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/detector.py]: 太长了，换行写，行首顶4个空格

File: paddlex/cv/models/detector.py [hunk 15]
```diff
@@ -347,6 +382,16 @@ def evaluate(self,
             return scores
 
     def predict(self, img_file, transforms=None):
+        """
+        Do inference.
+        Args:
+            img_file(list or str): Input image paths or arrays in BGR format.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/detector.py]: img_file补充说明list含义

File: paddlex/cv/models/detector.py [hunk 16]
```diff
@@ -248,9 +271,21 @@ def train(self,
...
-                 batch_size,
+                 batch_size=1,
                  metric=None,
                  return_details=False):
+        """
+        Evaluate the model.
+        Args:
+            eval_dataset(paddlex.dataset): Evaluation dataset.
+            batch_size(int, optional): Total batch size among all workers used for evaluation. Defaults to 1.
+            metric({'VOC', 'COCO', None}, optional): Evaluation metric. If None, determine the metric according to the dataset format. Defaults to None.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/detector.py]: 太长了，换行写，行首顶4个空格

File: paddlex/cv/models/detector.py [hunk 17]
```diff
@@ -347,6 +382,16 @@ def evaluate(self,
             return scores
 
     def predict(self, img_file, transforms=None):
+        """
+        Do inference.
+        Args:
+            img_file(list or str): Input image paths or arrays in BGR format.
+            transforms(paddlex.transforms.Compose or None, optional): Transforms for inputs. If None, the transforms for evaluation process will be used. Defaults to None.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/detector.py]: 太长了，换行写，行首顶4个空格

File: paddlex/cv/models/segmenter.py [hunk 18]
```diff
@@ -161,6 +161,25 @@ def train(self,
...
+        Args:
+            num_epochs(int): The number of epochs.
+            train_dataset(paddlex.dataset): Training dataset.
+            train_batch_size(int, optional): Total batch size among all workers used in training. Defaults to 2.
+            eval_dataset(paddlex.dataset, optional): Evaluation dataset. If None, the model will not be evaluated furing training process. Defaults to None.
+            optimizer(paddle.optimizer.Optimizer or None, optional): Optimizer used in training. If None, a default optimizer is used. Defaults to None.
+            save_interval_epochs(int, optional): Epoch interval for saving the model. Defaults to 1.
+            log_interval_steps(int, optional): Step interval for printing training information. Defaults to 10.
+            save_dir(str, optional): Directory to save the model. Defaults to 'output'.
+            pretrain_weights(str or None, optional): None or name/path of pretrained weights. If None, no pretrained weights will be loaded. Defaults to 'IMAGENET'.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/segmenter.py]: 太长了，换行写，行首顶4个空格

File: paddlex/cv/models/segmenter.py [hunk 19]
```diff
@@ -203,6 +222,17 @@ def train(self,
             use_vdl=use_vdl)
 
     def evaluate(self, eval_dataset, batch_size=1, return_details=False):
+        """
+        Evaluate the model.
+        Args:
+            eval_dataset(paddlex.dataset): Evaluation dataset.
+            batch_size(int, optional): Total batch size among all workers used for evaluation. Defaults to 1.
+            return_details(bool, optional): Whether to return evaluation details. Defaults to False.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/segmenter.py:233]: 说明都有哪些key以及key代表的含义

File: paddlex/cv/models/segmenter.py [hunk 20]
```diff
@@ -283,6 +313,16 @@ def evaluate(self, eval_dataset, batch_size=1, return_details=False):
...
 
     def predict(self, img_file, transforms=None):
+        """
+        Do inference.
+        Args:
+            img_file(list or str): Input image paths or arrays in BGR format.
+            transforms(paddlex.transforms.Compose or None, optional): Transforms for inputs. If None, the transforms for evaluation process will be used. Defaults to None.
+
+        Returns:
+            dict or list of dict: The prediction results.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/segmenter.py]: 说明都有哪些key以及key代表的含义

File: paddlex/cv/models/segmenter.py [hunk 21]
```diff
@@ -283,6 +313,16 @@ def evaluate(self, eval_dataset, batch_size=1, return_details=False):
         return eval_metrics
 
     def predict(self, img_file, transforms=None):
+        """
+        Do inference.
+        Args:
+            img_file(list or str): Input image paths or arrays in BGR format.
+            transforms(paddlex.transforms.Compose or None, optional): Transforms for inputs. If None, the transforms for evaluation process will be used. Defaults to None.
```
Comment (FlyingQianMM) [inline paddlex/cv/models/segmenter.py]: 补充说明img_file
太长了

File: paddlex/cv/transforms/batch_operators.py [hunk 22]
```diff
@@ -52,21 +52,30 @@ def __call__(self, samples):
...
     """
-    Resize image to target size randomly. random target_size and interpolation method
+    Resize a batch of input to random sizes.
+
+    Attention：If interp is 'RANDOM', the interpolation method will be chose randomly.
+
     Args:
-        target_size (list): image target size, must be list of (int or list)
-        interp (int): the interpolation method
+        target_sizes (List[int], List[list or tuple] or Tuple[lsit or tuple]): Multiple target sizes, each target size is an int or list/tuple.
```
Comment (FlyingQianMM) [inline paddlex/cv/transforms/batch_operators.py]: 加上Tuple[int], lsit -> list
解释int和list的区别

File: paddlex/cv/transforms/operators.py [hunk 23]
```diff
@@ -84,10 +85,11 @@ def __call__(self, sample):
...
 class Compose(Transform):
     """
-        Apply a series of data augmentation to the input.
-        All input images are in Height-Width-Channel ([H, W, C]) format.
+    Apply a series of data augmentation to the input.
+    All input images are in Height-Width-Channel ([H, W, C]) format.
+
     Args:
-        transforms (list of [paddlex.transforms.Transform]): List of data augmentations.
+        transforms (List[paddlex.transforms.Transform]): List of data augmentations.
```
Comment (FlyingQianMM) [inline paddlex/cv/transforms/operators.py]: augmentations -> preprocess or augmentations

File: paddlex/cv/transforms/batch_operators.py [hunk 24]
```diff
@@ -77,8 +86,28 @@ def __call__(self, samples):
 
 
 class BatchRandomResizeByShort(Transform):
+    """Resize a batch of input to random sizes with keeping the aspect ratio.
+
+    Attention：If interp is 'RANDOM', the interpolation method will be chose randomly.
+
+    Args:
+        short_sizes (List[int]): Target sizes of the shorter side of the image(s).
```
Comment (FlyingQianMM) [inline paddlex/cv/transforms/batch_operators.py]: + tuple

File: paddlex/cv/transforms/operators.py [hunk 25]
```diff
@@ -196,16 +200,16 @@ def apply(self, sample):
...
 
     - If target_size is an int，resize the image(s) to (target_size, target_size).
     - If target_size is a list or tuple, resize the image(s) to target_size.
     Attention：If interp is 'RANDOM', the interpolation method will be chose randomly.
 
     Args:
-        target_size (int/list/tuple): Target size.
-        interp ({'NEAREST', 'LINEAR', 'CUBIC', 'AREA', 'LANCZOS4', 'RANDOM'}): Interpolation method of resize,
-        'LINEAR' is used by default.
+        target_size (int, List[int] or Tuple[int]): Target size.
```
Comment (FlyingQianMM) [inline paddlex/cv/transforms/operators.py]: 这里说明下list/tuple时对应(height, width)

File: paddlex/cv/transforms/operators.py [hunk 26]
```diff
@@ -298,14 +302,14 @@ def apply(self, sample):
...
+    """
+    Resize input to random sizes.
 
     Attention：If interp is 'RANDOM', the interpolation method will be chose randomly.
 
     Args:
-        target_sizes (list): Multiple target sizes, each target size is an int or list/tuple.
-        interp ({'NEAREST', 'LINEAR', 'CUBIC', 'AREA', 'LANCZOS4', 'RANDOM'}): Interpolation method of resize,
-        'LINEAR' is used by default.
+        target_sizes (List[int], List[list or tuple] or Tuple[lsit or tuple]): Multiple target sizes, each target size is an int or list/tuple.
```
Comment (FlyingQianMM) [inline paddlex/cv/transforms/operators.py]: 同BatchRandomResize.

File: paddlex/cv/transforms/operators.py [hunk 27]
```diff
@@ -579,13 +585,14 @@ def apply(self, sample):
...
     Args:
-        mean(list/tuple): Mean of input image(s).
-        std(list/tuple): Standard deviation of input image(s).
-        min_val(list/tuple): Minimum value of input image(s).
-        max_val(list/tuple): Max val8ue of input image(s).
+        mean(List[float] or Tuple[float], optional): Mean of input image(s). Defaults to [0.485, 0.456, 0.406].
+        std(List[float] or Tuple[float], optional): Standard deviation of input image(s). Defaults to [0.229, 0.224, 0.225].
+        min_val(List[float] or Tuple[float], optional): Minimum value of input image(s). Defaults to [0, 0, 0, ].
+        max_val(List[float] or Tuple[float], optional): Max value of input image(s). Defaults to [255., 255., 255.].
     """
```
Comment (FlyingQianMM) [inline paddlex/cv/transforms/operators.py:599]: + is_scale

File: paddlex/cv/transforms/operators.py [hunk 28]
```diff
@@ -579,13 +585,14 @@ def apply(self, sample):
 
 
 class Normalize(Transform):
-    """Apply min-max normalization to the image(s) in input.
+    """
+    Apply min-max normalization to the image(s) in input.
```
Comment (FlyingQianMM) [inline paddlex/cv/transforms/operators.py:588]: 简单说明下步骤

File: paddlex/cv/transforms/operators.py [hunk 29]
```diff
@@ -628,12 +635,13 @@ def apply(self, sample):
...
+    """
+    Crop the input at the center.
     1. Locate the center of the image.
-    2. Crop the image.
+    2. Crop the sample.
 
     Args:
-        crop_size(int): target size of the cropped image(s), 224 by default.
+        crop_size(int, optional): target size of the cropped image(s). Defaults to 224.
     """
```
Comment (FlyingQianMM) [inline paddlex/cv/transforms/operators.py:648]: 这里为啥都加上了(s)？

File: paddlex/cv/models/base.py (hunk 1 above)
Comment (will-jl944) [reply paddlex/cv/models/base.py]: Done

File: paddlex/cv/models/base.py (hunk 2 above)
Comment (will-jl944) [reply paddlex/cv/models/base.py]: Done

File: paddlex/cv/models/classifier.py (hunk 3 above)
Comment (will-jl944) [reply paddlex/cv/models/classifier.py]: Done

File: paddlex/cv/models/classifier.py (hunk 4 above)
Comment (will-jl944) [reply paddlex/cv/models/classifier.py]: Done

File: paddlex/cv/models/classifier.py (hunk 5 above)
Comment (will-jl944) [reply paddlex/cv/models/classifier.py]: Done

File: paddlex/cv/models/classifier.py (hunk 6 above)
Comment (will-jl944) [reply paddlex/cv/models/classifier.py]: Done

File: paddlex/cv/models/detector.py (hunk 7 above)
Comment (will-jl944) [reply paddlex/cv/models/detector.py]: Done

File: paddlex/cv/models/classifier.py (hunk 8 above)
Comment (will-jl944) [reply paddlex/cv/models/classifier.py]: Done

File: paddlex/cv/models/detector.py (hunk 9 above)
Comment (will-jl944) [reply paddlex/cv/models/detector.py:291]: Done

File: paddlex/cv/models/detector.py (hunk 10 above)
Comment (will-jl944) [reply paddlex/cv/models/detector.py:407]: Done

File: paddlex/cv/models/classifier.py (hunk 12 above)
Comment (will-jl944) [reply paddlex/cv/models/classifier.py]: Done

File: paddlex/cv/models/classifier.py (hunk 13 above)
Comment (will-jl944) [reply paddlex/cv/models/classifier.py]: Done

File: paddlex/cv/models/detector.py (hunk 14 above)
Comment (will-jl944) [reply paddlex/cv/models/detector.py]: Done

File: paddlex/cv/models/detector.py (hunk 15 above)
Comment (will-jl944) [reply paddlex/cv/models/detector.py]: Done

File: paddlex/cv/models/detector.py (hunk 16 above)
Comment (will-jl944) [reply paddlex/cv/models/detector.py]: Done

File: paddlex/cv/models/detector.py (hunk 17 above)
Comment (will-jl944) [reply paddlex/cv/models/detector.py]: Done

File: paddlex/cv/models/segmenter.py (hunk 18 above)
Comment (will-jl944) [reply paddlex/cv/models/segmenter.py]: Done

File: paddlex/cv/models/segmenter.py (hunk 19 above)
Comment (will-jl944) [reply paddlex/cv/models/segmenter.py:233]: Done

File: paddlex/cv/models/segmenter.py (hunk 20 above)
Comment (will-jl944) [reply paddlex/cv/models/segmenter.py]: Done

File: paddlex/cv/models/segmenter.py (hunk 21 above)
Comment (will-jl944) [reply paddlex/cv/models/segmenter.py]: Done

File: paddlex/cv/transforms/batch_operators.py (hunk 22 above)
Comment (will-jl944) [reply paddlex/cv/transforms/batch_operators.py]: Done

File: paddlex/cv/transforms/operators.py (hunk 23 above)
Comment (will-jl944) [reply paddlex/cv/transforms/operators.py]: Done

File: paddlex/cv/transforms/batch_operators.py (hunk 24 above)
Comment (will-jl944) [reply paddlex/cv/transforms/batch_operators.py]: Done

File: paddlex/cv/transforms/operators.py (hunk 25 above)
Comment (will-jl944) [reply paddlex/cv/transforms/operators.py]: Done

File: paddlex/cv/transforms/batch_operators.py (hunk 22 above)
Comment (will-jl944) [reply paddlex/cv/transforms/batch_operators.py]: 加了See Also

File: paddlex/cv/transforms/operators.py (hunk 26 above)
Comment (will-jl944) [reply paddlex/cv/transforms/operators.py]: Done

File: paddlex/cv/transforms/operators.py (hunk 27 above)
Comment (will-jl944) [reply paddlex/cv/transforms/operators.py:599]: Done

File: paddlex/cv/transforms/operators.py (hunk 28 above)
Comment (will-jl944) [reply paddlex/cv/transforms/operators.py:588]: Done

File: paddlex/cv/transforms/operators.py (hunk 29 above)
Comment (will-jl944) [reply paddlex/cv/transforms/operators.py:648]: 因为输入可能是单张或者多张图片。
https://github.com/PaddlePaddle/PaddleX/blob/2ee50b10844e6b0ecf55546e6ce4c634c27d5b06/paddlex/cv/transforms/operators.py#L76

**Pull request merged**